src/onboard_flash.rs

```diff
@@ -343,6 +351,10 @@ impl OnboardFlash {
         self.scan();
         self.unlock_blocks();
     }
+    pub fn init_async_buf(&mut self) {
+        let mut flash_payload_buf = [0x42u8; 2115];
```

Comment:
I realise this code isn't being used at the moment, but there's no way this can work.  You're taking a reference to a stack variable that's only valid in the stack frame of this function, and then unsafely saying that it lives for the lifetime of the program!  For this to work and be safe, that stack allocated array needs to be passed in from the top stack frame of the program, that doesn't get popped off the stack until the whole program terminates.